### PR TITLE
[scaffolder] - Support filtering out of `steps` and `fields` based on `featureFlags`

### PR DESCRIPTION
**Pull request by tperehinets**
## Hey, I just made a Pull Request!

Issue: #18739

I fixed `featureFlag` in `scaffolder/next`.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by tperehinets** [comment]
@camilaibs and @benjdlambert , thanks for the feedback in my previous pull request! My previous branch was causing issues with `concurrently`, so I decided to create another one.

File: plugins/scaffolder-react/src/next/components/Workflow/Workflow.tsx [hunk 1]
```diff
@@ -77,6 +84,64 @@ export const Workflow = (workflowProps: WorkflowProps): JSX.Element | null => {
 
   const { loading, manifest, error } = useTemplateParameterSchema(templateRef);
 
+  const featureFlagKey = 'backstage:featureFlag';
+  const featureFlagApi = useApi(featureFlagsApiRef);
+
+  // return manifest with updates steps
+  const filterOutStepProperties = useCallback(() => {
```
Comment (benjdlambert) [inline plugins/scaffolder-react/src/next/components/Workflow/Workflow.tsx:91]: Can we move this out into a hook somewhere else other than this component?

Maybe in the `../../hooks` directory.

Something like `useFeatureFlaggedProperties()` or something?

File: plugins/scaffolder-react/src/next/components/Workflow/Workflow.tsx [hunk 2]
```diff
@@ -23,10 +23,17 @@ import {
...
+  errorApiRef,
+  useApi,
+  featureFlagsApiRef,
+} from '@backstage/core-plugin-api';
 import { useTemplateParameterSchema } from '../../hooks/useTemplateParameterSchema';
 import { Stepper, type StepperProps } from '../Stepper/Stepper';
 import { SecretsContextProvider } from '../../../secrets/SecretsContext';
+import cloneDeep from 'lodash/cloneDeep';
+import { useMemo, useCallback } from 'react';
+// import { clone } from 'lodash/fp';
```
Comment (benjdlambert) [inline plugins/scaffolder-react/src/next/components/Workflow/Workflow.tsx:36]: remove this.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Thanks for this! Just a few comments - also would be good to add some tests for the hook and the Workflow component changes.

**Comment by benjdlambert** [comment]
@tperehinets is there anything we can help with here? :pray:

**Comment by tperehinets** [comment]
> @tperehinets is there anything we can help with here? 🙏

Apologies for the delay, and thank you very much for your feedback. I've made the required adjustments. However, I have found that the hook `useTemplateSchema` is supposed to filter out properties that are not enabled with the feature flag. But, it does not filter the steps. Does that mean that it's better to enable sorting in `useTemplateSchema`, or implement a `useFeatureFlaggedProperties` hook as you recommend?

File: plugins/scaffolder-react/alpha-api-report.md [hunk 3]
```diff
@@ -288,6 +288,19 @@ export interface TemplateGroupsProps {
   templateFilter?: (entity: TemplateEntityV1beta3) => boolean;
 }
 
+// @alpha
+export const useFeatureFlaggedProperties: (
```
Comment (benjdlambert) [inline plugins/scaffolder-react/alpha-api-report.md:292]: `useFilteredSchemaProperties` or something? Maybe don't have to mention feature flags in the hook itself actually. Thinking it's a little cleaner as to what it's doing.

**Comment by benjdlambert** [comment]
@tperehinets  do you have time to fix this up? :pray: Would also be pretty good to add some tests too? :pray:

**Review by benjdlambert — APPROVED** [review]
Thanks for this! And the patience @tperehinets :tada:

**Review by dagda1 — APPROVED** [review]
Nice work!  This is definitely needed

**Comment by tperehinets** [comment]
@benjdlambert thank you so much for the feedback!

**Comment by benjdlambert** [comment]
@tperehinets could you look at some of these test failures? Not really sure what's going on here.

**Comment by benjdlambert** [comment]
@tperehinets nevermind I think I got it - want to get this in before the release today :tada:

**Pull request merged**